.github/workflows/release.yml

```diff
@@ -1,158 +1,206 @@
 name: release
 
+on:
+  workflow_dispatch:
+    inputs:
+      dry_run:
+        description: "Run the release without actually releasing bits"
+        type: boolean
+        default: true
+      crate:
+        description: "The crate to release. (containerd-shim-wasmtime)"
+        type: string
+      version:
+        description: "The version of the crate to release. (e.g., 1.2.3)"
+        type: string
+
 concurrency:
-  group: release-${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  group: release-${{ github.workflow }}-${{ inputs.crate }}-${{ inputs.version }}
   cancel-in-progress: true
 
-on:
-  push:
-    tags:
-      - '**/v[0-9]+.[0-9]+.*'
-
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  parse:
-    runs-on: ubuntu-latest
-    name: Parse ref
+  pre-release:
+    name: pre-release checks
+    runs-on: "ubuntu-latest"
     outputs:
-      crate: ${{ steps.parse.outputs.crate }}
-      version: ${{ steps.parse.outputs.version }}
-      runtime: ${{ steps.parse.outputs.runtime }}
+      crate: ${{ inputs.crate }}
+      runtime: ${{ steps.runtime_sub.outputs.runtime }}
+      version: ${{ inputs.version }}
+      ### is_shim is a string, not a boolean, so use: is_shim == 'true'
+      is_shim: ${{ steps.runtime_sub.outputs.is_shim }}
     steps:
-      - uses: actions/checkout@v3
+      # - name: Fail if branch is not main
+      #   if: github.event_name == 'workflow_dispatch' && github.ref != 'refs/heads/main'
+      #   run: |
+      #     echo "This workflow should not be triggered with workflow_dispatch on a branch other than main"
+      #     exit 1
+
+      ### Determine the name of the runtime and if it is a binary release or crates.io
+      - name: substring runtime
+        id: runtime_sub
+        uses: actions/github-script@v7
         with:
-          ref: ${{ github.ref }}
-      - id: parse
-        name: Parse ref
-        shell: bash
-        run: ./scripts/parse_ref.sh ${{ github.ref }} >> ${GITHUB_OUTPUT}
+          script: |
+            const crate = '${{ inputs.crate }}';
+            let runtime = crate.replace(/^containerd-shim-/, '');
+            core.setOutput('runtime', runtime);
```

Comment:
runtime seems a little mis-leading here since some of the crates aren't runtimes.  

 Another way to do this might be flip this logic to be hardcode the `shim_crates` instead of the `non_shim_crates` and set runtime if its included?

Reply:
`runtime` is a carry over from previous nomenclature. I couldn't think of much else other than runtime since it is mostly used in that way.

I'm indifferent about `shim_crates` vs. `non_shim_crates`. Do you feel like one reads easier than another? I fear my feeling for it has dulled from being too close to it.

Reply:
how about `project`?  I see how it is being used below, in some cases its the `runtime` others its the `crate`.  could be better to leave as is so we don't introduce another terminology 🤷